visvis2/material/_triangle.py

```diff
@@ -1,7 +1,11 @@
 from ._base import Material
 
 import wgpu  # only for flags/enums
-from python_shader import python2shader, RES_INPUT, RES_OUTPUT
+from python_shader import python2shader, RES_INPUT, RES_OUTPUT, RES_UNIFORM
+from python_shader import Struct, vec3
```

Comment:
The only concern I have is that this struct is tightly coupled to python-shader. We are going to lean on it very heavily I guess but we should keep in mind that it should remain possible for users to bring their own shaders. They might want to use glsl2spirv type shaders for example.

Reply:
Not necessarily. The only requirement (but we should formalize this some more, if this is the way we want to go) is that `.uniforms` is a ctypes object. Instantiating the `Struct(color=vec3)` is just a convenient way to create such a ctypes object.

And if numpy is a dependency, it might make more sense to use a numpy object instead.

Reply:
So then why exactly do you need to import it from python shader?

Reply:
Convenience. It allows to declare the type once, and use it both in the shader and for the `.uniforms` object. So they are guaranteed to match up. But your questions may suggest it's a bit too magical, perhaps. Maybe it would help to make it more explicit, e.g. `uniform_type.instantiate_ctypes_object()`?

Reply:
Let's keep it for now and see where it takes us :)